u256/src/division.rs

```diff
@@ -134,28 +134,21 @@ pub fn divrem_nbym(numerator: &mut [u64], divisor: &mut [u64]) {
             numerator[j + i] = a;
             borrow = b;
         }
-        let negative = numerator[j + n] < borrow;
-        // In release mode we don't bother computing numerator[j + n] because
-        // it should always come out zero.
-        if cfg!(debug_assertions) {
-            numerator[j + n] = numerator[j + n].wrapping_sub(borrow);
-        }
 
         // D5. Test remainder for negative result.
-        if negative {
+        if numerator[j + n] < borrow {
             // D6. Add back. (happens rarely)
             let mut carry = 0;
             for i in 0..n {
                 let (a, b) = adc(numerator[j + i], divisor[i], carry);
                 numerator[j + i] = a;
                 carry = b;
             }
-            if cfg!(debug_assertions) {
-                numerator[j + n] = numerator[j + n].wrapping_add(carry);
-            }
             qhat -= 1;
+            debug_assert_eq!(numerator[j + n].wrapping_sub(borrow).wrapping_add(carry), 0);
```

Comment:
From this debug statement it's not clear that `numerator[j + n].wrapping_sub(borrow).wrapping_add(carry)` would have been the updated value of `numerator[j + n]`.